### PR TITLE
TPROD-291 Remove use of KernelInterface::getName() and the kernel.name parameter

### PR DESCRIPTION
**Pull request by rohitpavaskar**
…y name i.e. app

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [n]
| New feature/enhancement? (use the a.x branch)      | [n]
| Deprecations?                          | [y]
| BC breaks? (use the c.x branch)        | [n ]
| Automated tests included?              | [n] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | https://mautic.atlassian.net/browse/TPROD-291

#### Description:
KernelInterface::getName() and the kernel.name parameter is removed from symfony 5 and we are removing use of the same.

#### Steps to test this PR:
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Try executing following command 'bin/console --version’. and check if outputs correctly and not breaking anything.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11138"></a>

### CODE REVIEW

**Review by RCheesley — DISMISSED** [review]
Command works fine:

```
gitpod /workspace/mautic (TPROD-291) $ ddev exec bin/console --version
Mautic 4.2.2 - app/prod (env: prod, debug: false)
```

**Review by volha-pivavarchyk — DISMISSED** [review]
Works well

image

**Comment by RCheesley** [comment]
This is good to go pending code review.

**Review by biozshock — DISMISSED** [review]

**Review by mollux — CHANGES_REQUESTED** [review]
This re-introduces a deprecated method. IMO the correct way is to set it statically to `app`

File: app/assets/scaffold/files/console [hunk 1]
```diff
@@ -38,5 +38,5 @@ $output = (false !== $input->getParameterOption(['--no-interaction', '-n'])) ? n
 $kernel      = new AppKernel($env, $debug);
 $application = new Application($kernel);
 $application->setName('Mautic');
-$application->setVersion($kernel->getVersion().' - '.$kernel->getName().'/'.$kernel->getEnvironment().($kernel->isDebug() ? '/debug' : ''));
+$application->setVersion($kernel->getVersion().' - '.basename($kernel->getRootDir()).'/'.$kernel->getEnvironment().($kernel->isDebug() ? '/debug' : ''));
```
Comment (mollux) [inline app/assets/scaffold/files/console]:
```suggestion
$application->setVersion($kernel->getVersion().' - app/'.$kernel->getEnvironment().($kernel->isDebug() ? '/debug' : ''));
```

File: bin/console [hunk 2]
```diff
@@ -38,5 +38,5 @@ $output = (false !== $input->getParameterOption(['--no-interaction', '-n'])) ? n
 $kernel      = new AppKernel($env, $debug);
 $application = new Application($kernel);
 $application->setName('Mautic');
-$application->setVersion($kernel->getVersion().' - '.$kernel->getName().'/'.$kernel->getEnvironment().($kernel->isDebug() ? '/debug' : ''));
+$application->setVersion($kernel->getVersion().' - '.basename($kernel->getRootDir()).'/'.$kernel->getEnvironment().($kernel->isDebug() ? '/debug' : ''));
```
Comment (mollux) [inline bin/console]:
```suggestion
$application->setVersion($kernel->getVersion().' - app/'.$kernel->getEnvironment().($kernel->isDebug() ? '/debug' : ''));
```

**Review by mollux — CHANGES_REQUESTED** [review]
`$kernel->getRootDir()` is deprecated, so using this method as a solution is not a good idea.
See https://github.com/mautic/mautic/pull/11145 and https://github.com/symfony/symfony/issues/29110

In https://github.com/mautic/mautic/pull/11145 the general approach is to change `$kernel->getRootDir()` with the string `app`.

**Review by escopecz — APPROVED** [review]

**Pull request merged**